rls/src/main/java/io/grpc/rls/RlsProtoConverters.java

```diff
@@ -188,9 +210,24 @@ static GrpcKeyBuilder convert(Map<String, ?> keyBuilder) {
       if (constantKeys == null) {
         constantKeys = ImmutableMap.of();
       }
+      checkUniqueKey(nameMatchers, constantKeys.keySet());
       return new GrpcKeyBuilder(names, nameMatchers, extraKeys, constantKeys);
     }
   }
 
+  private static void checkUniqueKey(List<NameMatcher> nameMatchers, Set<String> constantKeys) {
+    Set<String> keys = new HashSet<>();
+    keys.addAll(constantKeys);
+    keys.add("host");
+    keys.add("service");
+    keys.add("method");
+    for (NameMatcher nameMatcher :  nameMatchers) {
+      keys.add(nameMatcher.getKey());
+    }
```

Comment:
Would it make sense to exit early if the key is found? Something like

```suggestion
    for (NameMatcher nameMatcher :  nameMatchers) {
      if (!keys.add(nameMatcher.getKey()) {
        throw new IllegalArgumentException("keys in KeyBuilder must be unique");
      }
    }
```

This potentially can combined with the `throw new IllegalArgumentException(...` below, with a `boolean dup` + break in the for, and then `if (dup or keys.size() ... `.

Reply:
I also thought about exiting early. It seems to need multiple duplicate `if` and `throw` statements in that case.